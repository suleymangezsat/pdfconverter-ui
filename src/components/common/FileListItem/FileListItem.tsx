import {
  Avatar,
  Box,
  createStyles,
  ListItem,
  ListItemAvatar,
  ListItemSecondaryAction,
  ListItemText,
  makeStyles,
  Typography,
} from "@material-ui/core";
import FolderIcon from "@material-ui/icons/Folder";
import { memo } from "react";
import { formatBytes } from "../../../services/@formatters";

const useStyles = makeStyles(() =>
  createStyles({
    inline: {
      display: "inline",
    },
  })
);

type Props = {
  children: React.ReactNode;
  name: string;
  size: number;
  status: string;
  color?: "inherit" | "primary" | "secondary" | "action" | "disabled" | "error";
  createdAt?: Date;
  message?: string;
};

export const FileListItem = memo(
  ({ children, name, size, status, color, createdAt, message }: Props) => {
    const classes = useStyles();
    return (
      <ListItem divider>
        <ListItemAvatar>
          <Avatar>
            <FolderIcon color={color} />
          </Avatar>
        </ListItemAvatar>
        <ListItemText
          primary={name}
          secondaryTypographyProps={{ component: "div" }}
          secondary={
            <>
              <Box>
                <Typography
                  component="span"
                  variant="body2"
                  className={classes.inline}
                  color="textSecondary"
                >
                  {formatBytes(size)}
                </Typography>
                {createdAt && (
                  <Typography
                    component="span"
                    variant="body2"
                    className={classes.inline}
                    color="textSecondary"
                  >
                    {` Â· ${new Date(createdAt).toTimeString()}`}
                  </Typography>
                )}
              </Box>
              <Box>
                <Typography variant="body2" color="textSecondary">
                  {`Status : ${status}`}
                </Typography>
                {message && (
                  <Typography variant="body2" color="textSecondary">
                    {`Reason : ${message}`}
                  </Typography>
                )}
              </Box>
            </>
          }
        />
        <ListItemSecondaryAction>{children}</ListItemSecondaryAction>
      </ListItem>
    );
  }
);
