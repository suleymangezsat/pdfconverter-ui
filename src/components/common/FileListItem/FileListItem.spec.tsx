import React from "react";
import { FileListItem } from "./FileListItem";
import { shallow } from "enzyme";

describe("components/common/FileListItem", () => {
  const name = "filename.pdf";
  const size = 128000;
  const status = "INIT";
  const color = "error";
  const createdAt = new Date("10/11/2011");
  const message = "An error occured.";

  const prepareComponent = (props?: any) => (
    <FileListItem
      name={name}
      size={size}
      status={status}
      color={color}
      createdAt={createdAt}
      message={message}
      {...props}
    >
      {<div data-testid="test">Example</div>}
    </FileListItem>
  );

  it("should render FileListItem component properly", () => {
    const component = shallow(prepareComponent());
    expect(component).toMatchSnapshot();
  });
});
