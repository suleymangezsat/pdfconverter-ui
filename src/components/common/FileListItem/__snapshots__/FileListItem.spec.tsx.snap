// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/FileListItem should render FileListItem component properly 1`] = `
<WithStyles(ForwardRef(ListItem))
  divider={true}
>
  <WithStyles(ForwardRef(ListItemAvatar))>
    <WithStyles(ForwardRef(Avatar))>
      <Memo(ForwardRef(FolderIcon))
        color="error"
      />
    </WithStyles(ForwardRef(Avatar))>
  </WithStyles(ForwardRef(ListItemAvatar))>
  <WithStyles(ForwardRef(ListItemText))
    primary="filename.pdf"
    secondary={
      <React.Fragment>
        <Styled(MuiBox)>
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-inline-1"
            color="textSecondary"
            component="span"
            variant="body2"
          >
            125 KB
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-inline-1"
            color="textSecondary"
            component="span"
            variant="body2"
          >
             Â· 00:00:00 GMT+0300 (GMT+03:00)
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
        <Styled(MuiBox)>
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            variant="body2"
          >
            Status : INIT
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            variant="body2"
          >
            Reason : An error occured.
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
      </React.Fragment>
    }
    secondaryTypographyProps={
      Object {
        "component": "div",
      }
    }
  />
  <WithStyles(ForwardRef(ListItemSecondaryAction))>
    <div
      data-testid="test"
    >
      Example
    </div>
  </WithStyles(ForwardRef(ListItemSecondaryAction))>
</WithStyles(ForwardRef(ListItem))>
`;
